openvdb/Types.h

```diff
@@ -6,7 +6,14 @@
 
 #include "version.h"
 #include "Platform.h"
+
+#ifdef OPENVDB_USE_HALF
 #include <OpenEXR/half.h>
+#define OPENVDB_THROW_IF_HALF_NOT_SUPPORTED() (void) // Nop if OPENVDB_USE_HALF is enabled.
```

Comment:
I went with this approach to avoid lots of `#ifdef` lines and to have the error string in one place. Alternately this could be a function in the `openvdb` namespace, or I could go with just `OPENVDB_THROW(IoError, "OpenVDB has been compiled without Half support")` wherever it's needed.